### PR TITLE
#7690 First Playwright regression checklist test

### LINKED ISSUE #7690: First Playwright test

### PR DESCRIPTION
**Pull request by mnholtz**
## What does this PR do?

- Closes #7690 
- Adds a playwright test for activating the `@pixies/highlight-text` mod and asserting that it runs properly after visiting a page

## Discussion

- Originally, I had started writing this test for the `@pixies/giphy/giphy-search` mod, but this required 1) interacting with the quickbar, 2) interacting with a modal form, 3) interacting with the sidebar and 4) mocking an http request. I decided to dump that work in favor of this simpler mod.
    - However, I did retain learnings from going down that rabbit hole, which include learnings from Playwright with closed & open shadow root, iframes, and keyboard interaction to trigger the quick bar

## Checklist

- [x] Add tests
- [x] New files added to `src/tsconfig.strictNullChecks.json` (if possible)
- [x] Designate a primary reviewer @fungairino

### CODE REVIEW

File: end-to-end-tests/tests/extensionConsoleActivation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  baseURL,
+}) => {
+  const modName = "Highlight Specific Keywords When Page Loads";
+  const modId = "@pixies/highlight-keywords";
+
+  await page.goto(
+    `chrome-extension://${extensionId}/options.html#/marketplace/activate/${encodeURIComponent(
+      modId,
+    )}`,
+  );
```
Comment (fungairino) [inline end-to-end-tests/tests/extensionConsoleActivation.spec.ts:32]: nit: would make a helper method for getting the url for activating a specific modId.

as we continue adding tests that activate a specific mod, we should consider adding a fixture for installing a specific mod

File: end-to-end-tests/tests/extensionConsoleActivation.spec.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  await expect(page.getByText("Activate Mod")).toBeVisible();
+  await expect(page.getByText(modName)).toBeVisible();
+  await page.click("button:has-text('Activate')");
+  await expect(page.getByText(`Installed ${modName}`)).toBeVisible();
+
+  await page.goto(`${baseURL}/bootstrap-5`);
```
Comment (fungairino) [inline end-to-end-tests/tests/extensionConsoleActivation.spec.ts:39]: nit: I believe you can write `  await page.goto("/bootstrap-5");`

File: end-to-end-tests/tests/extensionConsoleActivation.spec.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { test, expect } from "../fixtures/extensionBase";
+
+test("can activate a mod with config options", async ({
```
Comment (fungairino) [inline end-to-end-tests/tests/extensionConsoleActivation.spec.ts:20]: nit: would rename this test and test file to be more specific (something like "activate `highlight-keywords` mod and test using it")

**Review by fungairino — APPROVED** [review]

File: end-to-end-tests/tests/extensionConsoleActivation.spec.ts (hunk 3 above)
Comment (mnholtz) [reply end-to-end-tests/tests/extensionConsoleActivation.spec.ts:20]: The goal for some of these tests is to e.g. "activate a mod with no options config", "activate a mod with built-in integrations", "activate a mod with required service configurations", etc. 

Do you have any thoughts about maintaining the high-level intention in the name?

File: end-to-end-tests/tests/extensionConsoleActivation.spec.ts (hunk 1 above)
Comment (twschiller) [reply end-to-end-tests/tests/extensionConsoleActivation.spec.ts:31]: > would make a helper method for getting the url for activating a specific modId.

You could potentially use our existing utils (which would redirect through the app though)? https://github.com/pixiebrix/pixiebrix-extension/blob/8bb5dbbdb6be2d2693e2b3de60a9b213cd29117a/src/activation/activationLinkUtils.ts#L40-L40

File: end-to-end-tests/tests/extensionConsoleActivation.spec.ts (hunk 3 above)
Comment (fungairino) [reply end-to-end-tests/tests/extensionConsoleActivation.spec.ts]: Yea, I like something like "activate and use a simple mod with no configuration"

**Pull request merged**